test_agent/rust/src/testagent.rs

```diff
@@ -0,0 +1,293 @@
+
+            let action = json_msg["action"].clone();
+            let json_data_value = json_msg["data"].clone();
+            let test_id = json_msg["test_id"].clone();
+
+            let json_str_ref = action
+                .as_str()
+                .expect("issue in converting value to string");
+
+            dbg!(json_str_ref);
+            let status = match json_str_ref {
+                constants::SEND_COMMAND => {
+                    let wu_message: WrapperUMessage =
+                        serde_json::from_value(json_data_value).unwrap(); // convert json to UMessage
+                    let u_message = wu_message.0;
+                    action_str = constants::SEND_COMMAND;
+                    self.utransport.send(u_message).await
+                }
+
+                constants::REGISTER_LISTENER_COMMAND => {
+                    let cloned_listener = Arc::clone(&arc_self);
+
+                    let wu_uuri: WrapperUUri = serde_json::from_value(json_data_value).unwrap(); // convert json to UMessage
+                    let u_uuri = wu_uuri.0;
+                    action_str = constants::REGISTER_LISTENER_COMMAND;
+                    self.utransport
+                        .register_listener(
+                            u_uuri,
+                            Arc::clone(&cloned_listener) as Arc<dyn UListener>,
+                        )
+                        .await
+                }
+
+                constants::UNREGISTER_LISTENER_COMMAND => {
+                    let cloned_listener = Arc::clone(&arc_self);
+                    let wu_uuri: WrapperUUri = serde_json::from_value(json_data_value).unwrap(); // convert json to UMessage
+                    let u_uuri = wu_uuri.0;
+                    action_str = constants::UNREGISTER_LISTENER_COMMAND;
+                    self.utransport
+                        .unregister_listener(
+                            u_uuri,
+                            Arc::clone(&cloned_listener) as Arc<dyn UListener>, /*&cloned_listener*/
+                        )
+                        .await
+                }
+
+                _ => Ok(()),
+            };
+
+            let mut status_dict: HashMap<String, _> = HashMap::new();
+
+            match status {
+                Ok(()) => {
+                    let status = UStatus::default();
+                    status_dict.insert(
+                        "message".to_string(),
+                        status.message.clone().unwrap_or_default(),
+                    );
+                    status_dict.insert("code".to_string(), 0.to_string());
+                }
+                Err(u_status) => {
+                    status_dict.insert(
+                        "message".to_string(),
+                        u_status.message.clone().unwrap_or_default(),
+                    );
+                    let enum_number = match u_status.get_code() {
```

Comment:
Hmmm :thinking: 

Is there some particular mapping you're trying to preserve here?

Wouldn't we be able to use u_status.get_code().value() to get the i32 representation of the code if that's what you're after?

Reply:
It seems like you're okay with the same ordering as normal, and if so, [see suggestion here](https://github.com/eclipse-uprotocol/up-tck/pull/23/files#r1581459354).